mm2src/coins/lp_coins.rs

```diff
@@ -2085,7 +2085,8 @@ pub trait MarketCoinOps {
     /// Signs raw utxo transaction in hexadecimal format as input and returns signed transaction in hexadecimal format
     async fn sign_raw_tx(&self, args: &SignRawTransactionRequest) -> RawTransactionResult;
 
-    fn wait_for_confirmations(&self, input: ConfirmPaymentInput) -> Box<dyn Future<Item = (), Error = String> + Send>;
+    /// Wait for confirmations and return the number of confirmed blocks
+    fn wait_for_confirmations(&self, input: ConfirmPaymentInput) -> Box<dyn Future<Item = u64, Error = String> + Send>;
```

Comment:
I am fine with returning number of confirmations here, so you can leave it. But it wasn't needed IMHO. As I can see it's only used here https://github.com/KomodoPlatform/komodo-defi-framework/blob/165ddfe8b19650da676c327b960d899cbb62c606/mm2src/mm2_main/src/lp_swap/taker_swap.rs#L1878 and we could have used the confirmations passed/used in `ConfirmPaymentInput` instead.

Just a note, but let's leave it :)

Reply:
We will revert it https://github.com/KomodoPlatform/komodo-defi-framework/pull/2199#discussion_r1777085347